provisioning/service/src/Contract/ContractApiHttp.cs

```
@@ -185,15 +185,26 @@ internal class ContractApiHttp : IContractApiHttp
 
         private static void ValidateHttpResponse(ContractApiResponse response)
         {
-            if (response.StatusCode >= HttpStatusCode.InternalServerError ||
-                (int)response.StatusCode == 429)
+            if (response.Body == null)
             {
-                throw new ProvisioningServiceClientHttpException(response, isTransient: true);
+                throw new DeviceProvisioningServiceException(response.ErrorMessage, response.StatusCode, response.Fields);
             }
-            else if (response.StatusCode >= HttpStatusCode.Ambiguous)
+
+            try
             {
-                throw new ProvisioningServiceClientHttpException(response, isTransient: false);
+                ResponseBody responseBody = JsonConvert.DeserializeObject<ResponseBody>(response.Body);
+
+                if (response.StatusCode >= HttpStatusCode.Ambiguous)
+                {
+                    throw new DeviceProvisioningServiceException(
+                        $"{response.ErrorMessage}:{response.Body}",
+                        response.StatusCode,
+                        responseBody.ErrorCode,
+                        responseBody.TrackingId,
+                        response.Fields);
+                }
             }
+            catch (JsonException) { }
```

So, if we can't parse, we just silently fail? That doesn't seem right.